lib/maths/CDataFrameUtils.cc

```diff
@@ -322,149 +359,188 @@ CDataFrameUtils::TDoubleVec CDataFrameUtils::micWithColumn(const CColumnValue& t
         return zeros;
     }
 
-    auto method = frame.inMainMemory() ? micWithColumnDataFrameInMemory
-                                       : micWithColumnDataFrameOnDisk;
+    auto method = frame.inMainMemory() ? metricMicWithColumnDataFrameInMemory
+                                       : metricMicWithColumnDataFrameOnDisk;
 
-    return method(target, frame, columnMask,
+    return method(target, frame, rowMask, columnMask,
                   std::min(NUMBER_SAMPLES_TO_COMPUTE_MIC, frame.numberRows()));
 }
 
 bool CDataFrameUtils::isMissing(double x) {
     return CMathsFuncs::isFinite(x) == false;
 }
 
-CDataFrameUtils::TSizeDoublePrVecVec
-CDataFrameUtils::categoryMicWithColumnDataFrameInMemory(const CColumnValue& target,
-                                                        std::size_t numberThreads,
-                                                        const core::CDataFrame& frame,
-                                                        const TSizeVec& columnMask,
-                                                        std::size_t numberSamples,
-                                                        double minimumFrequency) {
+CDataFrameUtils::TSizeDoublePrVecVecVec CDataFrameUtils::categoricalMicWithColumnDataFrameInMemory(
```

Comment:
This function seems long. Would it be possible to break down in sub-functions to make reading and undertanding it easier?

Reply:
I factored out a couple of pieces that can be shared which cuts it down a bit (and reduces code duplication). Otherwise, you end up passing and modifying local variables in the different sub-functions, which I think would make it harder to understand what is going on. See [this](https://github.com/elastic/ml-cpp/pull/591/commits/3982962d91b31d8d0f2950683820a3563cce1b49) commit.